lib/api/bulk.js

```diff
@@ -221,9 +245,37 @@ Job.prototype.list = function(callback) {
     batchInfoList = _.isArray(batchInfoList.batchInfo) ? batchInfoList.batchInfo : [ batchInfoList.batchInfo ];
     return batchInfoList;
   }).thenCall(callback);
-
 };
 
+Job.prototype.retrieveBatches = function(callback) {
+  var self = this;
+  return self.list().then((batchInfoList) => {
+    return Promise.all(batchInfoList.map(function(batchInfo) {
+      if (batchInfo.state === 'NotProcessed' || parseInt(batchInfo.numberRecordsProcessed, 10) === 0) return Promise.resolve();
+      if (batchInfo.state === 'Failed' && parseInt(batchInfo.numberRecordsProcessed, 10) === 0) {
+        throw new Error(batchInfo.stateMessage)
+      }
+      return self.batch(batchInfo.id).retrieve();
+    }));
+  }).then(function(batchResults) {
+    const results = [];
+    batchResults.forEach(function(batchResult) {
+      if (!batchResult) return;
+      results.push(...batchResult);
+    });
+    return results;
+  }).then(function(results) {
+    return self.close().then(function() {
+      return results;
+    })
+  }).fail(function(err) {
+    self.emit("error", err);
+    self.abort().then(() => {
```

Comment:
Should this be returned? Kinda weird for the error to occur after the callback is called, otherwise; also this error will be an unhandled rejection at the moment.

Reply:
I've added
```
.fail(function(err) {
  self.emit('error', err);
})
```

Though i was not sure how to simulate the relevant flow